dandi/delete.py

```diff
@@ -246,5 +247,8 @@ def find_local_asset(filepath: str) -> tuple[str, str]:
 
 
 def is_same_url(url1: str, url2: str) -> bool:
-    # TODO: Use a real URL library like furl, purl, or yarl
-    return url1.rstrip("/") == url2.rstrip("/")
+    u1 = URL(url1)
+    u1 = u1.with_path(u1.path.rstrip("/"))
+    u2 = URL(url2)
+    u2 = u2.with_path(u2.path.rstrip("/"))
+    return u1 == u2
```

Comment:
so how is this better? will it reorder query params or smth to find equality? if so -- would be nice to add a testcase which would solidify adding such capability which was not present before.

Reply:
It normalizes various URL bits, such as lowercasing the scheme & host and normalizing internationalized domain names.

Reply:
Tests added: e58df625